src/app/CommandResponseSender.h

```diff
@@ -124,11 +124,14 @@ class CommandResponseSender : public Messaging::ExchangeDelegate
     }
 
     /**
-     * @brief Sets final non-path specific StatusResponse error after sending all InvokeResponseMessages
+     * @brief Queues a single, non-path-specific, error to be sent after all queued InvokeRequestMessages.
```

Comment:
So here's what I am not following:

1. Where do InvokeRequestMessages come here at all?  We're not queueing those here.
2. Why would we want to send any other messages at all if we know we have a non-path-specific (i.e. "this whole thing failed") error to send?

This logic is just not making sense to me.

Reply:
1. Queued InvokeResponseMessages (I recognize the typo, sorry) comes from `AddInvokeResponseToSend`

2. We have the InvokeResponseMessages, why not send them back? The application might want to know what happened to some of the commands before it failed.